### PR TITLE
Adjust iframe height to account for embedded content margins

### PR DESCRIPTION
**Pull request by jayjay-w**
## Description

Resolve an issue where embedded content in pender-iframe gets cut off due to incorrect height calculations. This issue was particularly noticeable with Facebook and TikTok posts, which were offset by `16px` and `18px` respectively, corresponding to the margins inside the embedded content.

To address this, an adjustable padding (adjustment) of 20px has been added to the calculated height in the `setHeight` case of the message Callback function. Additionally, the height value is parsed as an integer to ensure accurate calculations.

These changes aim to ensure that the iframe height fully accommodates the embedded content, providing consistent rendering across platforms. Room has been left for future platform-specific adjustments if necessary.

References: CV2-5742

## How has this been tested?

Confirmed that the iframe is not being clipped.

## Checklist

- [x] I have performed a self-review of my own code
- [ ] I have added unit and feature tests, if the PR implements a new feature or otherwise would benefit from additional testing
- [ ] I have added regression tests, if the PR fixes a bug
- [ ] I have added logging, exception reporting, and custom tracing with any additional information required for debugging
- [ ] I considered secure coding practices when writing this code. Any security concerns are noted above.
- [ ] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [ ] My changes generate no new warnings
- [ ] If I added a third party module, I included a rationale for doing so and followed our current [guidelines](https://meedan.atlassian.net/wiki/spaces/ENG/overview#Choose-the-%E2%80%9Cright%E2%80%9D-3rd-party-module)

### CODE REVIEW

**Review by vasconsaurus — APPROVED** [review]

**Comment by vasconsaurus** [comment]
There are some errors, but they look like they are coming from my changes from yesterday 🤔 I'll take a look.

**Review by caiosba — APPROVED** [review]
Thanks for adding a test we didn't have!

**Comment by caiosba** [comment]
@jayjay-w please delete `test/models/.DS_Store` - you pushed it by mistake.

File: test/models/parser/facebook_item_test.rb [hunk 1]
```diff
@@ -540,5 +540,26 @@ def apify_error_response
...
+
+  test "html_for_facebook_post includes script for height adjustment" do
+    username = "test_user"
+    request_url = "https://www.facebook.com/test_user/posts/12345"
+    html_page = Nokogiri::HTML("<html></html>") # Simulate a valid HTML page
+  
+    parser = Parser::FacebookItem.new(request_url)
+  
+    parser.stub(:not_an_event_page, true) do
+      parser.stub(:not_a_group_post, true) do
```
Comment (caiosba) [inline test/models/parser/facebook_item_test.rb:553]: Interesting, I didn't know this block-style syntax,

**Comment by jayjay-w** [comment]
Thank you @vasconsaurus and @caiosba, merging!

**Pull request merged**